From 977af23447c25f327dea22784dbce6cde88023b7 Mon Sep 17 00:00:00 2001
From: Socrates Wei <socrates@dniserver.dnish.net>
Date: Wed, 2 Apr 2014 15:30:27 +0800
Subject: [PATCH] forked-daapd: do not check pairing guid

For fit NTGR SPEC rev12 16.15.4
6. Once allow a device to control music play via Apple Remote, the client can
always access the iTunes server even after Remote app restart or router reboot.

Signed-off-by: Socrates Wei <socrates@dniserver.dnish.net>
---
 src/httpd_daap.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/httpd_daap.c b/src/httpd_daap.c
index 3a24b94..b193ea6 100644
--- a/src/httpd_daap.c
+++ b/src/httpd_daap.c
@@ -64,6 +64,7 @@ extern struct event_base *evbase_httpd;
 /* Update requests refresh interval in seconds */
 #define DAAP_UPDATE_REFRESH  0
 
+#define DONT_CHECK_PAIRING_GUID
 
 struct uri_map {
   regex_t preg;
@@ -794,6 +795,8 @@ daap_reply_login(struct evhttp_request *req, struct evbuffer *evbuf, char **uri,
       pi.guid = strdup(guid + 2); /* Skip leading 0X */
 
       ret = db_pairing_fetch_byguid(&pi);
+#ifdef DONT_CHECK_PAIRING_GUID
+#else
       if (ret < 0)
 	{
 	  DPRINTF(E_LOG, L_DAAP, "Login attempt with invalid pairing-guid\n");
@@ -802,6 +805,7 @@ daap_reply_login(struct evhttp_request *req, struct evbuffer *evbuf, char **uri,
 	  evhttp_send_error(req, 403, "Forbidden");
 	  return;
 	}
+#endif
 
       DPRINTF(E_INFO, L_DAAP, "Remote '%s' logging in with GUID %s\n", pi.name, pi.guid);
       free_pi(&pi, 1);
-- 
1.7.2.5

